content/documentation/architecture.adoc

```diff
@@ -0,0 +1,116 @@
+applications.
+
+The following diagram illustrates the components involved in Kiali and its
+interactions:
+
+image::architecture.png[Kiali architecture]
+
+== Kiali backend
+
+Tha backend is the application that runs in the container application platform.
+It's written in http://golang.org/[Go]. The code can be found at
+https://github.com/kiali/kiali[kiali/kiali GitHub repository].
+
+This is the component that communicates with Istio parts, retrieves and
+processes data, and send processed data to the frontend.
+
+The backend doesn't need storage. When running the backend locally, all
+configurations are set in files and environment variables (including the user
+and password used for authentication). When deploying the backend to a cluster
+using the docker images, configmaps and secrets are used to set the
+configurations.
+
+== Kiali frontend
+
+The frontend is the application that runs in the browser.  It's a React
+application, written in Typescript. The code can be found at
+https://github.com/kiali/kiali-ui[kiali/kiali-ui GitHub repository].
+
+In a standard deployment, the backend provides the frontend, but once the
+browser has downloaded the frontend, it will run mainly in the browser and only
+assets will be requested. The frontend queries the Kiali backend in order to
+get data and present it to the user. 
+
+Since, currently, there are no options for personalizations, it's mainly
+stateless. Some data may be persisted - like session credentials -, but this
+data is stored in the browser and won't be available in other browsers nor
+other devices.
+
+== Istio
+
+Istio is a Kiali requirement. It's the component that provides and controls the
+service mesh. Although Kiali and Istio can be installed separately, Kiali
+depends on Istio and won't be useful if it's not present.
+
+Kiali doesn't communicate with Istio. This is the reason the diagram shows a
+dashed line. Communication between Kiali and Istio occurs through the cluster
+API.
+
+== Prometheus
+
+Prometheus is an Istio dependency. When Istio telemetry is enabled, metrics
+data is stored in Prometheus. Kiali uses the data stored in Prometheus to
+figure out the mesh topology, show metrics, calculate health, show possible
+problems, etc.
+
+Kiali communicates directly with Prometheus. This effectively turns Prometheus
+into a Kiali dependency. Istio telemetry must be installed for most Kiali
+features to work properly. 
+
+Currently, Kiali relies on
+https://istio.io/docs/reference/config/policy-and-telemetry/metrics/[Istio's
+default metrics] set. Make sure that these default metrics configuration are
+always in place. Custom metrics are not supported by Kiali. Should you need
+custom metrics or a variant of the default ones, create new Istio metrics using
+custom names.
+
+== Cluster API
+
+Kiali uses the API of the container application platform in order to fetch
+workload definitions, and to fetch Istio configurations.
```

Comment:
isn't line 81-82 very similar to what is said in line 84 and onwards?

Reply:
Yes, a little. I wanted to explain further. I changed the first paragraph to avoid repetition, but I'm not sure if the new text works.
Please check. If it still sounds redundant, I could merge paragraphs.